sdk/compiler/damlc/daml-stdlib-src/DA/Assert.daml

```diff
@@ -59,3 +62,19 @@ assertBeforeMsg msg time = do
   if time < now
     then assertFail msg
     else return ()
+
+-- | Check whether the ledger time of the transaction is at or past the given deadline.
+-- If it's not, abort with a message.
+assertWithinDeadline: Text -> Time -> Update ()
+assertWithinDeadline name deadline = do
+  deadlineExceeded <- isLedgerTimeGE deadline
+  when deadlineExceeded $ assertFail $
+    "Ledger time is at or past deadline '" <> name <> "' at " <> show deadline
```

Comment:
Let's check what message is raised from a failure of this. I remember that assertion failures have a pure UX as they typically report like a double negation.

Here we'd just like to get `Interpretation error: stdlib.daml.com/deadline-exceeded: ledger time is at or past deadline '<name>' at <date>`.

Note that `stdlib.daml.com/deadline-exceeded` is the error-id. I expect you can raise this using the new tools introduced by @samuel-williams-da

Reply:
Use `stdlib.daml.com/deadline-not-exceeded` for the converse function below.

Reply:
@samuel-williams-da thanks for that. That wasn't an obvious set of things to find and use with submit - is there any documentation on this anywhere?